include/ui/pixel-display-painted-item.hpp

```diff
@@ -0,0 +1,509 @@
+* at Technical University of Munich
+*
+* This program is free software: you can redistribute it and/or modify
+* it under the terms of the GNU General Public License as published by
+* the Free Software Foundation, either version 3 of the License, or
+* (at your option) any later version.
+*
+* This program is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with this program. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#ifndef ERAGPSIM_UI_PIXEL_DISPLAY_PAINTED_ITEM_HPP
+#define ERAGPSIM_UI_PIXEL_DISPLAY_PAINTED_ITEM_HPP
+
+#include <QImage>
+#include <QPainter>
+#include <QQuickPaintedItem>
+#include <QRect>
+#include <cstdint>
+#include <functional>
+#include <memory>
+#include <utility>
+#include <vector>
+
+#include "common/optional.hpp"
+//#include "core/memory-access.hpp"
+//#include "ui/output-component.hpp"
+class OutputComponent;
+class MemoryValue;
+
+// just for testing reasons
+#define PSIZE 16
+#define CMODE 1
+
+namespace colorMode {
+struct ColorMode;
+struct Options {
+  // address of of the pixelBuffer(pointer)
+  std::size_t pixelBaseAddress = 0;
+  // address of of the colorBuffer(pointer)
+  std::size_t colorBaseAddress = 4;
+  // width of the image
+  std::size_t width = PSIZE;
+  // height of the image
+  std::size_t height = PSIZE;
+  // colorMode RGB:Monochrome
+  std::size_t colorMode = CMODE;
+  // number of bits the red part of a color in RGB mode has
+  std::size_t rBit = 8;
+  // number of bits the green part of a color in RGB mode has
+  std::size_t gBit = 8;
+  // number of bits the blue part of a color in RGB mode has
+  std::size_t bBit = 8;
+  // TODO::use enums, maybe? probably not, cause qml is annoying
+  // false->row_major, true->columns->major
+  bool columns_rows = false;
+  // mirrors the image horizontally
+  bool horizontallyMirrored = false;// TODO
+  // mirrors the image vertically
+  bool verticallyMirrored = false;// TODO
+  // packs pixel data as tight as possible
+  bool tight = false;
+  // interpret PixelBaseAddress as pointer to the PixelBuffer
+  bool pixelBufferPointerLike = true;
+  // interpret ColorBaseAddress as pointer to the colorTable
+  bool colorTablePointerLike = true;
+  // ignore n bytes after each pixel (in RGB mode)
+  std::size_t freeBytes = 1;
+  // waste n bits in each byte (in Monochrome mode)
+  std::size_t freeBits = 0;
+
+  /*
+   * \brief returns the current ColorMode
+   */
+  ColorMode getColorMode() const;
+  /*
+   * \brief returns the color of the pixel at (x,y)
+   * \param memoryAccess source to load the data from
+   * \param x X position of the Pixel
+   * \param y Y position of the Pixel
+   */
+  std::uint32_t getPixel(Optional<OutputComponent *> memoryAccess,
```

Comment:
because afaik QImage uses 32bit colors, or at least there is no color Mode I know of with more than 32 bit